build_common.ps1

```diff
@@ -997,134 +788,241 @@ class BuildProject {
+				if (([int]$numErr) -gt 0) {
+					$clr = "Red"
+				} elseif (([int]$numWarn) -gt 0) {
+					$clr = "Yellow"
+				} else {
+					$clr = "Green"
+				}
+			} else {
+				if ($outTxt -Match "Error") {
+					$clr = "Red"
+				} else {
+					$clr = "Yellow"
+				}
+			}
+			Write-Host $outTxt -ForegroundColor $clr
+		} else {
+			Write-Host $outTxt
+		}
+	}
+
+	[void]Finish([int] $exitCode) {
+		([StdoutReceiver]$this).Finish($exitCode)
+	}
+}
+
+# TODO: Filter more lines for HL cook? `Hashing`? `SHA: package not found`? `Couldn't find localized resource`?
+# `Warning, Texture file cache waste exceeds`? `Warning, Package _ is not conformed`?
+class ModcookReceiver : StdoutReceiver {
+	[bool] $foundNativeScriptError = $false
+	[bool] $foundRelevantZeroExitError = $false
+	[bool] $lastLineWasAdding = $false
+	[bool] $permitAdditional = $false
+
+	[bool] $modCook
+
+	ModcookReceiver([bool] $modCook){
+		$this.modCook = $modCook
+	}
+	
+	[void]ParseLine([string] $outTxt) {
+		([StdoutReceiver]$this).ParseLine($outTxt)
+		$permitLine = $true # Default to true in case there is something we don't handle
+
+		if ($outTxt.StartsWith("Adding package") -or $outTxt.StartsWith("Adding level") -or $outTxt.StartsWith("Adding script") -or $outTxt.StartsWith("GFx movie package")) {
+			if ($outTxt.Contains("\Mods\")) {
+				$permitLine = $true
+			} else {
+				$permitLine = $false
+
+				if (!$this.lastLineWasAdding) {
+					Write-Host "[Adding sdk assets ...]"
+				}
+			}
+
+			$this.lastLineWasAdding = !$permitLine
+			$this.permitAdditional = $permitLine
+		} elseif ($outTxt.StartsWith("Adding additional")) {
+			$permitLine = $this.permitAdditional
+		} else {
+			$this.lastLineWasAdding = $false
+			$permitLine = $true
+		}
+
+		if ($permitLine) {
+			Write-Host $outTxt
+		}
+
+		if ($outTxt.StartsWith("Error")) {
+			# * OnlineSubsystemSteamworks and AkAudio cannot be removed from cook and generate 4 errors when mod is built in debug - needs to be ignored
+			if ($this.modCook -and ($outTxt.Contains("AkAudio") -or $outTxt.Contains("OnlineSubsystemSteamworks"))) {
+				$this.foundNativeScriptError = $true
+			} else {
+				$this.foundRelevantZeroExitError = $true
+			}
+		}
+	}
+
+	[void]Finish([int] $exitCode) {
+		# Do not call super because we may have a successful cook with a non-zero exit code
+		if ($this.crashDetected) {
+			ThrowFailure "Crash detected while $($this.processDescr)"
+		}
+		if ($this.foundNativeScriptError) {
+			Write-Host ""
+			Write-Host "Detected errors about AkAudio and/or OnlineSubsystemSteamworks - these are safe to ignore."
+			Write-Host "If you want to get rid of them, you would need to build the mod in non-debug mode"
+			Write-Host "at least once - the errors will then go away (until the BuildCache folder is cleared/deleted)"
+			Write-Host ""
+		}
+
+		if ($this.foundRelevantZeroExitError) {
+			ThrowFailure "Found a relevant error while cooking (commandlet exited successfully but logs indicate error)"
```

Comment:
> commandlet exited successfully

All errors cause non-0 exit, not just the `AkAudio` and `OnlineSubsystemSteamworks` ones. A 0 exit is a successful one, barring crashes